pkg/resource/apply.go

```diff
@@ -125,11 +123,29 @@ func (r *helper) CreateRuntimeObject(obj runtime.Object, scheme *runtime.Scheme)
 	return r.Create(data)
 }
 
+func (r *helper) fakeApply() {
+	// real world data indicates that for our slowest non-delete request type (POST):
+	// histogram_quantile(0.9, (sum without(controller,endpoint,instance,job,namespace,pod,resource,service,status)(rate(hive_kube_client_request_seconds_bucket{remote="true",controller="clustersync"}[2h]))))
+	//
+	// 50% of requests are under 0.027s
+	// 80% of requests are under 0.045s
+	// 90% of requests are under 0.053s
+	// 99% of requests are under 0.230s
+	in := []int{27, 27, 27, 27, 27, 45, 45, 45, 53, 230} // milliseconds
+	randomIndex := rand.Intn(len(in))
+	wait := time.Duration(in[randomIndex] * 1000000) // nanoseconds to match the duration unit
+	r.logger.WithField("sleepMillis", wait.Milliseconds()).Debug("sleeping to simulate an apply")
+	time.Sleep(wait)
+}
+
 func (r *helper) createOnly(f cmdutil.Factory, obj []byte) (ApplyResult, error) {
 	info, err := r.getResourceInternalInfo(f, obj)
 	if err != nil {
 		return "", err
 	}
+	if info == nil {
+		r.logger.Debug("err getting info")
```

Comment:
Should likely return an error here. Is this something you've observed? (where err == nil and info == nil?)

Reply:
I honestly don't even remember adding this. I'll remove, probably something from when I was stuck trying to figure out what was wrong.